src/manifest.rs

```diff
@@ -95,6 +111,42 @@ impl Default for OciImageManifest {
     }
 }
 
+impl From<OciImageIndex> for OciManifest {
+    fn from(m: OciImageIndex) -> Self {
+        Self::ImageIndex(m)
+    }
+}
+impl From<OciImageManifest> for OciManifest {
+    fn from(m: OciImageManifest) -> Self {
+        Self::Image(m)
+    }
+}
+
+fn write_json(obj: &impl Serialize, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(
```

Comment:
If we do stick with using JSON for the `Display` implementation, I'd recommend we use `serde_json::to_writer` instead to make this as efficient as possible (it skips allocating a string and then writing it)

Reply:
Good point, I'll wait for the decision above and submit a change if JSON stays in.

Reply:
I looked into doing that, unfortunately `Formatter` is implementing the [`std::fmt::Write`](https://doc.rust-lang.org/std/fmt/trait.Write.html) trait, not the [`std::io::Write`](https://doc.rust-lang.org/nightly/std/io/trait.Write.html) one which is required by `serde_json::to_writer`.

We have to leave the code as it is